tests/contrib/vertexai/test_vertexai.py

```diff
@@ -0,0 +1,499 @@
+            "Why do bears hibernate?",
+            generation_config=vertexai.generative_models.GenerationConfig(stop_sequences=["x"], max_output_tokens=50, temperature=0),
+        )
+
+@pytest.mark.snapshot
+def test_vertexai_completion_stream(vertexai, mock_client):
+     with patch.object(vertexai.generative_models.GenerativeModel, '_prediction_client', new_callable=PropertyMock) as mock_client_property:
+        mock_client_property.return_value = mock_client
+        mock_client.responses["stream_generate_content"] = [
+            (_mock_completion_stream_chunk(chunk) for chunk in MOCK_COMPLETION_STREAM_CHUNKS)
+        ]
+        llm = vertexai.generative_models.GenerativeModel("gemini-1.5-flash")
+        response = llm.generate_content(
+            "How big is the solar system?",
+            generation_config=vertexai.generative_models.GenerationConfig(stop_sequences=["x"], max_output_tokens=30, temperature=1.0),
+            stream=True,
+        )
+        for _ in response:
+            pass
+
+@pytest.mark.snapshot(ignores=["meta.error.stack", "meta.error.message"])
+def test_vertexai_completion_stream_error(vertexai, mock_client):
+    with patch.object(vertexai.generative_models.GenerativeModel, '_prediction_client', new_callable=PropertyMock) as mock_client_property:
+        mock_client_property.return_value = mock_client
+        mock_client.responses["stream_generate_content"] = [
+            (_mock_completion_stream_chunk(chunk) for chunk in MOCK_COMPLETION_STREAM_CHUNKS)
+        ]
+        llm = vertexai.generative_models.GenerativeModel("gemini-1.5-flash")
+        with pytest.raises(TypeError):
+            response = llm.generate_content(
+                "How big is the solar system?",
+                generation_config=vertexai.generative_models.GenerationConfig(stop_sequences=["x"], max_output_tokens=30, temperature=1.0),
+                stream = True,
+                candidate_count=2, # candidate_count is not a valid keyword argument 
+            )
+            for _ in response:
+                pass
+
+
+@pytest.mark.snapshot
+def test_vertexai_completion_stream_tool(vertexai, mock_client):
+    with patch.object(vertexai.generative_models.GenerativeModel, '_prediction_client', new_callable=PropertyMock) as mock_client_property:
+        mock_client_property.return_value = mock_client
+        mock_client.responses["stream_generate_content"] = [
+            (_mock_completion_stream_chunk(chunk) for chunk in MOCK_COMPLETION_TOOL_CALL_STREAM_CHUNKS)
+        ]
+        llm = vertexai.generative_models.GenerativeModel("gemini-1.5-flash", tools=[weather_tool])
+        response = llm.generate_content(
+            "What is the weather like in New York City?",
+            generation_config=vertexai.generative_models.GenerationConfig(stop_sequences=["x"], max_output_tokens=30, temperature=1.0),
+            stream=True,
+        )
+        for _ in response:
+            pass
+
+@pytest.mark.snapshot
```

Comment:
This one's a nit but in the interest of minimizing test files, can we try re-using snapshot files when possible? Especially since we're reusing/testing the same data for sync/async versions of each test
```suggestion
@pytest.mark.snapshot(token="tests.contrib.vertexai.test_vertexai.test_vertexai_completion") // should rename the snapshot in the sync test as well
```

Reply:
@Yun-Kim i see you have immediately embraced being a go developer

Reply:
LOL 😂

Reply:
💀

Reply:
Thanks for the tip! I updated the test file to reuse the same snapshot for the completion and chat version of each test (this halves the number of snapshot files).